js/lib/hooks.js

```diff
@@ -6,24 +6,94 @@ elgg.provide('elgg.config.hooks');
 elgg.provide('elgg.config.instant_hooks');
 elgg.provide('elgg.config.triggered_hooks');
 
+!function() {
```

Comment:
Why not make it a named AMD module with all the hooks API inside and use require within `elgg.` functions? That way we will have a module we can extract into an anonymous module later if we have to.